src/mame/adp/servicetastatur.cpp

```diff
@@ -119,90 +144,136 @@ void servicet_state::servicet_map(address_map &map)
 
 void servicet_state::servicet_data(address_map &map)
 {
-	map(0x0000, 0xffff).rw(FUNC(servicet_state::bus_r), FUNC(servicet_state::bus_w));
+	map(0x0070, 0x0070).w(m_lcd, FUNC(hd44780_device::control_w));
+	map(0x0071, 0x0071).r(m_lcd, FUNC(hd44780_device::control_r));
+	map(0x0072, 0x0072).w(m_lcd, FUNC(hd44780_device::data_w));
+	map(0x0073, 0x0073).r(m_lcd, FUNC(hd44780_device::data_r));
+
+	map(0x0010, 0x0030).w(FUNC(servicet_state::gsg_w));
+	map(0x0040, 0x0050).r(FUNC(servicet_state::gsg_r));
+	map(0x0060, 0x0060).w(FUNC(servicet_state::gsg_w));
+	map(0x4000, 0x4000).w(FUNC(servicet_state::gsg_w));
 }
 
 static INPUT_PORTS_START( servicet )
 	PORT_START("IN0") // P1.0
-	PORT_BIT( 0x10, IP_ACTIVE_HIGH, IPT_KEYPAD) PORT_NAME("OK") PORT_CODE(KEYCODE_ENTER)
-	PORT_BIT( 0x20, IP_ACTIVE_HIGH, IPT_KEYPAD) PORT_NAME("F4") PORT_CODE(KEYCODE_F4)
-	PORT_BIT( 0x40, IP_ACTIVE_HIGH, IPT_JOYSTICK_UP)    PORT_4WAY
+	PORT_BIT( 0x01, IP_ACTIVE_HIGH, IPT_UNUSED)
+	PORT_BIT( 0x02, IP_ACTIVE_LOW, IPT_UNUSED)
+	PORT_BIT( 0x04, IP_ACTIVE_LOW, IPT_UNUSED)
+	PORT_BIT( 0x08, IP_ACTIVE_LOW, IPT_UNUSED)
+	PORT_BIT( 0x10, IP_ACTIVE_LOW, IPT_START) PORT_NAME("OK")
+	PORT_BIT( 0x20, IP_ACTIVE_LOW, IPT_BUTTON4) PORT_NAME("F4")
+	PORT_BIT( 0x40, IP_ACTIVE_LOW, IPT_JOYSTICK_UP)    PORT_4WAY
+	PORT_BIT( 0x80, IP_ACTIVE_LOW, IPT_UNUSED)
```

Comment:
Spacing is a bit messed up here. And you can shorten this to `PORT_BIT( 0x0f, IP_ACTIVE_LOW, IPT_UNUSED)`. "Random" `ACTIVE_HIGH` is weird too, just update the code used to read the inputs.